system/Database/MySQLi/Connection.php

```diff
@@ -505,17 +508,33 @@ protected function _foreignKeyData(string $table): array
         }
         $query = $query->getResultObject();
 
-        $retVal = [];
+        $ind = [];
 
         foreach ($query as $row) {
-            $obj                      = new stdClass();
-            $obj->constraint_name     = $row->CONSTRAINT_NAME;
-            $obj->table_name          = $row->TABLE_NAME;
-            $obj->column_name         = $row->COLUMN_NAME;
-            $obj->foreign_table_name  = $row->REFERENCED_TABLE_NAME;
-            $obj->foreign_column_name = $row->REFERENCED_COLUMN_NAME;
```

Comment:
Changed the data structure a bit. There was some inconsistency between DBMS. They have all been standardized.

Instead of returning multiple rows in the case of composite keys only a single row is returned. The column names will be an array of columns used in key/composite key.

The first loop creates the data structure. The second converts it into an object.